packages/select/src/components/select/multiSelect.tsx

```diff
@@ -178,17 +188,21 @@ export class MultiSelect<T> extends React.PureComponent<IMultiSelectProps<T>, IM
         Utils.safeInvoke(this.props.onQueryChange, query, evt);
     };
 
+    // Popover interaction kind is CLICK, so this only handles click events.
+    // Note that we defer to the next animation frame in order to get the latest document.activeElement
     private handlePopoverInteraction = (nextOpenState: boolean) =>
-        // deferring to rAF to get properly updated document.activeElement
         requestAnimationFrame(() => {
-            if (this.input != null && this.input !== document.activeElement) {
-                // the input is no longer focused so we can close the popover
+            const isInputFocused = this.input === document.activeElement;
+
+            if (this.input != null && !isInputFocused) {
+                // input is no longer focused, we should close the popover
                 this.setState({ isOpen: false });
             } else if (!this.props.openOnKeyDown) {
-                // open the popover when focusing the tag input
+                // we should open immediately on click focus events
                 this.setState({ isOpen: true });
             }
-            Utils.safeInvokeMember(this.props.popoverProps, "onInteraction", nextOpenState);
+
+            this.props.popoverProps?.onInteraction?.(nextOpenState);
```

Comment:
why is `onInteraction` not always treated the same as `setState`?

Reply:
there are some code paths here when `setState` does not get called. for example when the input gets clicked and `openOnKeyDown={true}`. to be "correct" according to the behavior of Popovers in general, we should still call `onInteraction`.